ooniprobe/AppDelegate.mm

```diff
@@ -143,6 +155,53 @@ -(void)application:(UIApplication *)application didReceiveLocalNotification:(UIL
     //TODO start your long running bg task here
 }
 
+- (BOOL)application:(UIApplication *)application handleOpenURL:(NSURL *)url {
+    //NSLog(@"url recieved: %@", url);
+    //NSLog(@"query string: %@", [url query]);
+    
+    //creating parameters dict
+    NSDictionary *dict = [DictionaryUtility parseQueryString:[url query]];
+    NSDictionary *parameters = [DictionaryUtility getParametersFromDict:dict];
+    
+    NSString *minimum_version = [parameters objectForKey:@"mv"];
+    if (minimum_version != nil){
+        if ([minimum_version compare:[[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"] options:NSNumericSearch] == NSOrderedDescending) {
+            //actualVersion is lower than the requiredVersion
+            UIAlertView *alertView = [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"ooniprobe_outdate", nil) message:NSLocalizedString(@"ooniprobe_outdate_msg", nil) delegate:nil cancelButtonTitle:NSLocalizedString(@"Cancel", nil) otherButtonTitles:NSLocalizedString(@"ok", nil), nil];
+            alertView.tag = 2;
+            [alertView show];
+        }
+        else {
+            NSString *action = [url host];
```

Comment:
@lorenzoPrimi is it correct to state that here you're taking the `host` of the URL because the URL starts like `ooni://nettest`? (Maybe it would help a comment on top of the routine stating what the URL format is, or it would also possibly do a link to the specification of the URLs)

Reply:
Uhm, from [run.ooni.io](https://run.ooni.io/) I _also_ see URIs like: `https://run.ooni.io/nettest?tn=ndt&mv=1.2.0`.

Reply:
The point is: there the host is `run.ooni.io`.

Reply:
Here the host is run.ooni.io when opened from the website, is nettest when opened from a link ooni://nettest.